module/blobs/blob_io.go

```diff
@@ -0,0 +1,305 @@
+package blobs
+
+import (
+	"errors"
+	"io"
+	"sync"
+)
+
```

Comment:
can you add a high level description of how these structs should be used and the rational?